acl/errors.go

```diff
@@ -117,12 +117,23 @@ func PermissionDenied(msg string, args ...interface{}) PermissionDeniedError {
 }
 
 // TODO Extract information from Authorizer
-func PermissionDeniedByACL(_ Authorizer, context *AuthorizerContext, resource Resource, accessLevel AccessLevel, resourceID string) PermissionDeniedError {
+func PermissionDeniedByACL(authz Authorizer, context *AuthorizerContext, resource Resource, accessLevel AccessLevel, resourceID string) PermissionDeniedError {
 	desc := NewResourceDescriptor(resourceID, context)
-	return PermissionDeniedError{Accessor: "", Resource: resource, AccessLevel: accessLevel, ResourceID: desc}
+	return PermissionDeniedError{Accessor: ExtractAccessorID(authz), Resource: resource, AccessLevel: accessLevel, ResourceID: desc}
 }
 
-func PermissionDeniedByACLUnnamed(_ Authorizer, context *AuthorizerContext, resource Resource, accessLevel AccessLevel) PermissionDeniedError {
+func PermissionDeniedByACLUnnamed(authz Authorizer, context *AuthorizerContext, resource Resource, accessLevel AccessLevel) PermissionDeniedError {
 	desc := NewResourceDescriptor("", context)
-	return PermissionDeniedError{Accessor: "", Resource: resource, AccessLevel: accessLevel, ResourceID: desc}
+	return PermissionDeniedError{Accessor: ExtractAccessorID(authz), Resource: resource, AccessLevel: accessLevel, ResourceID: desc}
+}
+
+func ExtractAccessorID(authz interface{}) string {
+	var accessor string
+	switch v := authz.(type) {
+	case AllowAuthorizer:
+		accessor = v.AccessorID
+	case string:
+		accessor = v
+	}
```

Comment:
Are there other authorizers that can exist here (eg. `ChainedAuthorizer`)? If so, that would result in an empty return value. Is this ok?

Reply:
Empty value is ok; the string conversion routine handles that gracefully.